### DISCUSSION TITLE
[solved] how to setup a quick preview from main branch

### DISCUSSION THREAD

**Discussion by atherdon** [atherdon/dash-back#21] [Ideas]
рано или поздно, но нам прийдется подумать о том, как и куда деплоить наш бекенд.

есть разные кейсы:
- например я сейчас немогу быстро поднять локальный проект чтобы посмотреть что все работает.
- или завтра у меня будет собрание с другими редакторами, и они попросят показать им быстро прогресс, который мы сделали по бекенду.

сейчас это займет определенное время. но если есть варианты с auto-deploy и т.д. можем попробывать\подумать о том, куда выкладывать.

тоесть если мы найдем интересный вариант как быстро поднимать копии проекта - очень сильно поможет.

сс @kolserdav

**Comment by kolserdav** [comment]
> рано или поздно, но нам прийдется подумать о том, как и куда деплоить наш бекенд.
> 
> есть разные кейсы:
> 
> * например я сейчас немогу быстро поднять локальный проект чтобы посмотреть что все работает.
> * или завтра у меня будет собрание с другими редакторами, и они попросят показать им быстро прогресс, который мы сделали по бекенду.
> 
> сейчас это займет определенное время. но если есть варианты с auto-deploy и т.д. можем попробывать\подумать о том, куда выкладывать.
> 
> тоесть если мы найдем интересный вариант как быстро поднимать копии проекта - очень сильно поможет.
> 
> сс @kolserdav

Предлагаю на Hetzner, вот моя реферальная ссылка https://hetzner.cloud/?ref=0Vh7MSaYpRFy по ней получаете 20 евро на счет.
Я сам без рефералов выбрал его, так как там хороший баланс цена/качество, сейчас там у меня серв за 3 евро в месяц, мне пока хватает.
А вам эти 20 евро для старта пригодятся, чтобы как раз разобраться как там инвойсы начисляются.

**Reply by kolserdav** [reply]
Если виртуальную машину поднимите, можно самую дешевую, пока нет нагрузок, в любой момент можно будет масштабировать (Centos 7 - самый экономный по ресурсам и самый стабильный, из тех images что там из коробки), при создании ssh ключи надо будет сделать. Подключитесь, пользователя создадите:
`useradd name`
Добавите его в группу sudo:
`usermod -aG sudo name` и мне датите его, я там поставлю nvm, node...
Пока руками будем из git а деплоить, а там разберемся.

**Reply by kolserdav** [reply]
Если при `useradd` не предложит ввести пароль, то еще надо будет `passwd name`

**Reply by kolserdav** [reply]
А если `git` сами поставите, то можно и без `sudo` я всё равно `nvm` локально ставить буду

**Reply by kolserdav** [reply]
Всё таки с `sudo`, так как ребутить надо будет (сам процесс почти не потребляет, но после работы `yarn` система память куда то резервирует, не разбирал ещё этот момент), да и nginx ещё настраивать и перезапускать плюс snapd для создания ssl сертификата (letsencrypt) , ещё файервол.

**Reply by atherdon** [reply]
Мне кажется мы как раз вчера с моим другом девопсом про hetzner разговаривали. это те немцы что сидят на основной магистрали интернета и у них дешевый хостинг - верно?

у нас сейчас есть аккаунт на linode - там крутится php сервер с которого мы шлем почтовую рассылку в которой я принимаю участие. еще у нас есть аккаунт на aws - там нету серверов, только instance который как раз и отправляет письма по запросу php сервера.

перенесу [часть про мелкие проекты](https://github.com/atherdon/dash-back/discussions/24#discussioncomment-351903) в отдельную ветку.

**Reply by kolserdav** [reply]
Да, дешевле я не видел.

**Comment by kolserdav** [comment]
Подождите, сейчас на сервере запущу, так быстрее будет чем с виндой воевать, потом...

**Reply by atherdon** [reply]
я тож так думаю. да, можем вначале у вас наверное стартануть - а потом подумаем о "правильном варианте". потому что проблема с деплоями у нас на всех проектах. и если мы сможем убить несколько птиц одним камнем - будет круто. я думал подключить моего знакомого DevOps - но его кинули на новый проект + у него сертификация по aws. но если он освободится в марте и настроит все для нас - это очень сильно облегчит мне жизнь

**Reply by kolserdav** [reply]
> я тож так думаю. да, можем вначале у вас наверное стартануть - а потом подумаем о "правильном варианте". потому что проблема с деплоями у нас на всех проектах. и если мы сможем убить несколько птиц одним камнем - будет круто. я думал подключить моего знакомого DevOps - но его кинули на новый проект + у него сертификация по aws. но если он освободится в марте и настроит все для нас - это очень сильно облегчит мне жизнь

с Linux у меня нет проблем)
У вас случайно нет Windows администратора?!))

**Reply by atherdon** [reply]
< с Linux у меня нет проблем)
У вас случайно нет Windows администратора?!))

винда отстой. просто для того чтобы поднять серв на локалке у Вадима - приходится придумывать обходные пути.
Я все еще считаю что вопрос deploy & preview нам нужно вынести в отдельный майлстоун и атаковать эту проблему. так будет гораздо проще

@kolserdav

**Reply by vadim9999** [reply]
Бек должен работать на любых ОС. Привязка к одной ОС и железу очень плохая идея

**Reply by vadim9999** [reply]
Хорошо тогда чтобы не тормозить процес буду на линуксе фигачить

**Comment by kolserdav** [comment]
Запустил на сервере,  `http://localhost...` в `postman.json` поменяйте на  `https://dash-back.uyem.ru/graphql`

**Reply by kolserdav** [reply]
Или лучше создать копию для remote, как вам удобней смотрите

**Reply by atherdon** [reply]
cc @vadim9999

**Comment by atherdon** [comment]
полностью согласен

>

**Comment by vadim9999** [comment]
@kolserdav what do you think about auth0? Now we have authentication with auth0 on frontend side. You can connect their api and we can send to backend token getted from auth0

**Reply by kolserdav** [reply]
I am hardly familiar with this

**Reply by atherdon** [reply]
we definitely need to explore it. because i'm lazy to build just-another-login-form for 20 users only. because login-forms never was an easy part, but always remind me of my first commercial coding experience
